tests/test_loss.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+import tensorflow as tf
+import numpy as np
+
+from zfit import ztf
+import zfit.core.basepdf
+from zfit.core.limits import Range
+from zfit.minimizers.minimizer_minuit import MinuitMinimizer
+import zfit.pdfs.dist_tfp
+from zfit.pdfs.dist_tfp import Normal
+from zfit.pdfs.basic import Gauss
+from zfit.core.parameter import FitParameter
+import zfit.settings
+from zfit.core.loss import unbinned_nll, UnbinnedNLL
+
+mu_true = 1.2
+sigma_true = 4.1
+
+test_values_np = np.random.normal(loc=mu_true, scale=sigma_true, size=1000)
+
+low, high = -24.3, 28.6
+mu1 = FitParameter("mu1", ztf.to_float(mu_true)-0.2, mu_true - 1., mu_true + 1.)
```

Comment:
Why do we need `ztf.to_float` here?

Reply:
Currently, Gauss actually expects a parameter and does not convert the value therefore. But this should be changed! You're right

Reply:
Or better, autoconverted to fixed parameters, right?

Reply:
That's a little bit the point here: it does not expect fixed parameters *per se* (it's a pdf in the end). But to use it as a pdf to constrain a value, we need fixed parameters.

So to be precise: whenever anything expects a *fixed/floating* parameter and get's a float, convert to a fixed param.

I agree on it!

Reply:
Ah yeah, I was sloppy: anything that expects a parameter and gets a float, is converted to a fixed parameter (unnamed?).

Reply:
unnamed -> somehow auto-named? Like "AUTOPARAM_fixed" or something like this?

Reply:
Do we need names? If so, autonamed then!